phoenix-server-client/pom.xml

```diff
@@ -26,29 +26,126 @@
 
   <properties>
     <top.dir>${project.basedir}/..</top.dir>
+    <shaded.package>org.apache.phoenix.shaded</shaded.package>
   </properties>
 
   <build>
     <plugins>
       <plugin>
-        <artifactId>maven-assembly-plugin</artifactId>
-        <executions>
-          <execution>
-            <id>thin-client</id>
-            <phase>package</phase>
-            <goals>
-              <goal>single</goal>
-            </goals>
-            <configuration>
-              <attach>false</attach>
-              <finalName>phoenix-${project.version}</finalName>
-              <descriptors>
-                <descriptor>src/build/thin-client.xml</descriptor>
-              </descriptors>
-            </configuration>
-          </execution>
-        </executions>
-      </plugin>
+      <artifactId>maven-assembly-plugin</artifactId>
+      <configuration>
+        <skipAssembly>true</skipAssembly>
+      </configuration>
+    </plugin>
+    <plugin>
+      <groupId>org.apache.maven.plugins</groupId>
+      <artifactId>maven-shade-plugin</artifactId>
+      <executions>
+        <execution>
+          <id>thin-client</id>
+          <phase>package</phase>
+          <goals>
+            <goal>shade</goal>
+          </goals>
+          <configuration>
+            <createSourcesJar>false</createSourcesJar>
+            <shadedArtifactAttached>false</shadedArtifactAttached>
+            <promoteTransitiveDependencies>true</promoteTransitiveDependencies>
+            <shadeTestJar>false</shadeTestJar>
+            <finalName>phoenix-${project.version}-thin-client</finalName>
+            <!--artifactSet>
```

Comment:
why commented out? We can remove if not needed anymore because it is enumerated below.